odoo/addons/base/models/res_users.py

```diff
@@ -1127,6 +1144,26 @@ def _remove_reified_groups(self, values):
             else:
                 values1[key] = val
 
+        portal = self.env.ref('base.group_portal',raise_if_not_found=False)
+        public = self.env.ref('base.group_public',raise_if_not_found=False)
+        if add:
+            #If we choose Portal or Public as user type, only this group is added.
+            if portal and portal.id in add:
+                add = [portal.id]
```

Comment:
This is wrong.
There are some usability groups that should be kept.

Reply:
The _remove_reified_groups is only use in the UsersView class, and it affects only the groups that are present as fields in the user view form. The implied groups are still added to the user, that means that the user in the end not only have the portal group but also the groups: Technical Settings / Addresses in Sales Orders; Technical Settings / Multi-website and Technical Settings / Tax display B2B.

Reply:
can a test be added to ensure that (if not already exists)

Reply:
I just added a test to ensure that the implied groups are added to the user.